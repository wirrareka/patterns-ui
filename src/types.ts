export type Selection = { [id: string]: boolean }

export type FetchResponse<T> = {
  items: T[],
  total: number
}

export type FilterComparator = 'contains' | 'starts_with' | 'ends_with' | 'eq' | 'lt' | 'gt' | 'lte' | 'gte' | 'range'

export type FilterValue = string | number | [number, number] | Date | [Date, Date] | undefined

export type FilterState = {
  column: string
  active: boolean
  value?: FilterValue
  comparator: FilterComparator
  type: ColumnType
}

export type FilterStates = {
  [id: string]: FilterState
}

export enum ColumnType {
  Text,
  Number,
  Date,
  DateTime,
  Custom
}
export interface Column<T> {
  id: string
  title: string
  type: ColumnType
  format?: (item: T) => React.ReactElement
  flex?: number
  width?: number
  sortable?: boolean
  filterable?: boolean
  visible?: boolean
}

export type PageSizeItem = { title: string, value: number }

export type Locale = {
  code: string
  range_from: string
  range_to: string
  day: string
  at_day: string
  older_than: string
  newer_than: string
  range: string
  value: string
  equal: string
  less_than: string
  greater_than: string
  contains: string
  starts_with: string
  ends_with: string
  per_page: string
  selection_none: string
  selection_one: string
  selection_few: string
  selection_many: string
  items_none: string
  items_one: string
  items_few: string
  items_many: string
}