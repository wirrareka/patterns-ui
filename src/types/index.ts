import Currency from '../models/currency'
import bigDecimal from 'js-big-decimal'

export { PatternSettings } from './pattern_settings'
export { Locale } from './locale'

export type FormatDateType = (date: Date) => string
export type ParseDateType = (date: string) => Date
export type FormatPriceType = (price: bigDecimal | number, currency?: Currency) => string

export type Selection = { [id: string]: boolean }

export type FetchResponse<T> = {
  items: T[],
  total: number
}

export type FilterComparator = 'contains' | 'starts_with' | 'ends_with' | 'eq' | 'lt' | 'gt' | 'lte' | 'gte' | 'range'

export type FilterValue = string | number | [number, number] | Date | [Date, Date] | undefined

export type FilterState = {
  column: string
  active: boolean
  value?: FilterValue
  comparator: FilterComparator
  type: ColumnType
}

export type FilterStates = {
  [id: string]: FilterState
}

export enum ColumnType {
  Text,
  Number,
  Date,
  DateTime,
  Custom
}

export enum Alignment {
  Center = 'center',
  Left = 'left',
  Right = 'right'
}

export interface Column<T> {
  id: string
  title: string
  type: ColumnType
  format?: (item: T) => React.ReactElement
  flex?: number
  width?: number
  sortable?: boolean
  filterable?: boolean
  visible?: boolean
  alignment?: Alignment
}

export type PageSizeItem = { title: string, value: number }

export interface PaymentMethod {
  id: string
  title: string
}
