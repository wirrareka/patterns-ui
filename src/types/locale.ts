export type Locale = {
  code: string
  range_from: string
  range_to: string
  day: string
  at_day: string
  older_than: string
  newer_than: string
  range: string
  value: string
  equal: string
  less_than: string
  greater_than: string
  contains: string
  starts_with: string
  ends_with: string
  per_page: string
  page: string
  selection_none: string
  selection_one: string
  selection_few: string
  selection_many: string
  items_none: string
  items_one: string
  items_few: string
  items_many: string
  first: string
  last: string
  datePosted: string
  dateDelivered: string
  dateDue: string
  paymentMethod: string
  bank: string
  bankAccount: string
  variableSymbol: string
  constantSymbol: string
  'newItem': string
  'invoice.new': string
  'invoice.type': string
  'offer.new': string
  'offer.type': string
  'order.new': string
  'order.type': string
  'payment_methods.cash': string
  'payment_methods.cash_on_delivery': string
  'payment_methods.card': string
  'payment_methods.bank_transfer': string
  'payment_methods.paypal': string
  name: string
  street: string
  street2: string
  zip: string
  city: string
  country: string
  businessId: string
  vatId: string
  vatPayerId: string
  vendor: string
  customer: string
  edit: string
  edit_code: string
  edit_note: string
  title: string
  quantity: string
  vat: string
  vatBase: string
  vatValue: string
  vatPrice: string
  vatRecap: string
  unitPriceNoVat: string
  linePriceNoVat: string
  linePriceWithVat: string
  signature: string
  total: string
  note: string
  delete: string,
  cancel: string
}
