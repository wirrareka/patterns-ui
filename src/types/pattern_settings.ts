import { Contactable } from "../interfaces"
import { Currency } from "../models"
import PaymentMethod from "../models/payment_method"
import { FormatDateType, ParseDateType, FormatPriceType } from '../types'

export type PatternSettings = {
  name: string
  logo: string
  bank: {
    name: string
    code: string
    swift: string
    account: string
  }
  company: Contactable
  currency: {
    defaultCurrency: Currency
    currencies: Currency[]
  }
  payment: {
    defaultMethod: PaymentMethod
    methods: PaymentMethod[]
  },
  format: {
    date: FormatDateType
    parseDate: ParseDateType
    price: FormatPriceType
  },
  invoices: {
    showVat: boolean
  },
  orders: {
    showVat: boolean
  }
}