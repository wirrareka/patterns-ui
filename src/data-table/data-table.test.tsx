import React from "react"
import { shallow } from "enzyme"

import DataTable from "./data-table"
import { Column, ColumnType, FilterStates } from './types'

class TestModel {
  id: string
  name: string

  constructor(id: string, name: string) {
    this.id = id
    this.name = name
  }
}

const TestTable = DataTable.ofType<TestModel>()



const items = [
  new TestModel("yolo1", "yolo 1"),
  new TestModel("yolo2", "yolo 2"),
  new TestModel("yolo3", "yolo 3"),
]

describe("DataTable", () => {
  let props: any

  beforeEach(() => {
    props = {
      expandable: false,
      sort: 'name',
      sortDir: 'asc',
      multiple: false,
      fetch: async (page: number, pageSize: number, sort: string, sortDir: 'asc' | 'desc', filters: FilterStates) => {
        return await new Promise((resolve, reject) => {
          resolve({ items, total: 0 })
        })
      },
      items: [],
      theme: "primary",
      columns: [
        {
          id: 'name',
          title: 'Meno',
          type: ColumnType.Text,
          format: (item: TestModel) => <span>{ item.name }</span>,
          sortable: true,
          filterable: true,
          visible: true
        },
      ] as Column<TestModel>[]
    }
  })

  const renderWrapper = () => shallow(<TestTable {...props} />)

  describe("Snapshots", () => {
    it("should match snapshots as primary themed", () => {
      expect(renderWrapper()).toMatchSnapshot()
    })
  })
})
