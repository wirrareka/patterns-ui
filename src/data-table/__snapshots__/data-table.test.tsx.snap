// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DataTable Snapshots should match snapshots as primary themed 1`] = `
<styled.div>
  <styled.div>
    <styled.div
      className="datatable-cell"
      head={true}
      key="datatable-header-name"
    >
      <div
        onClick={[Function]}
        style={
          Object {
            "flex": 3,
          }
        }
      >
        <span>
          Meno
        </span>
      </div>
      <div
        style={
          Object {
            "alignItems": "flex-end",
            "display": "flex",
            "flex": 1,
            "flexDirection": "row",
            "justifyContent": "flex-end",
          }
        }
      >
        <Blueprint3.Button
          className=""
          icon="chevron-up"
          intent="primary"
          minimal={true}
          onClick={[Function]}
        />
        <Blueprint3.Popover
          boundary="scrollParent"
          captureDismiss={false}
          defaultIsOpen={false}
          disabled={false}
          fill={false}
          hasBackdrop={false}
          hoverCloseDelay={300}
          hoverOpenDelay={150}
          inheritDarkTheme={true}
          interactionKind="click"
          minimal={false}
          modifiers={Object {}}
          openOnTargetFocus={true}
          position="bottom"
          targetTagName="span"
          transitionDuration={300}
          usePortal={true}
          wrapperTagName="span"
        >
          <Blueprint3.Button
            icon={
              <Blueprint3.Icon
                color="#A7B6C2"
                icon="filter"
              />
            }
            minimal={true}
            style={
              Object {
                "alignSelf": "center",
                "justifySelf": "center",
                "marginBottom": 0,
              }
            }
          />
          <FilterRenderer
            column={
              Object {
                "filterable": true,
                "format": [Function],
                "id": "name",
                "sortable": true,
                "title": "Meno",
                "type": 0,
                "visible": true,
              }
            }
            filterState={
              Object {
                "active": false,
                "column": "name",
                "comparator": "contains",
                "type": 0,
                "value": "",
              }
            }
            onFilterStateChange={[Function]}
          />
        </Blueprint3.Popover>
      </div>
    </styled.div>
  </styled.div>
  <styled.div>
    <styled.div
      key="data-table-empty-row-1"
    >
      <styled.div
        className="bp3-skeleton"
        key="data-table-item-empty-cell-name-1"
      >
         
      </styled.div>
    </styled.div>
    <styled.div
      key="data-table-empty-row-2"
    >
      <styled.div
        className="bp3-skeleton"
        key="data-table-item-empty-cell-name-2"
      >
         
      </styled.div>
    </styled.div>
    <styled.div
      key="data-table-empty-row-3"
    >
      <styled.div
        className="bp3-skeleton"
        key="data-table-item-empty-cell-name-3"
      >
         
      </styled.div>
    </styled.div>
    <styled.div
      key="data-table-empty-row-4"
    >
      <styled.div
        className="bp3-skeleton"
        key="data-table-item-empty-cell-name-4"
      >
         
      </styled.div>
    </styled.div>
    <styled.div
      key="data-table-empty-row-5"
    >
      <styled.div
        className="bp3-skeleton"
        key="data-table-item-empty-cell-name-5"
      >
         
      </styled.div>
    </styled.div>
    <styled.div
      key="data-table-empty-row-6"
    >
      <styled.div
        className="bp3-skeleton"
        key="data-table-item-empty-cell-name-6"
      >
         
      </styled.div>
    </styled.div>
    <styled.div
      key="data-table-empty-row-7"
    >
      <styled.div
        className="bp3-skeleton"
        key="data-table-item-empty-cell-name-7"
      >
         
      </styled.div>
    </styled.div>
    <styled.div
      key="data-table-empty-row-8"
    >
      <styled.div
        className="bp3-skeleton"
        key="data-table-item-empty-cell-name-8"
      >
         
      </styled.div>
    </styled.div>
    <styled.div
      key="data-table-empty-row-9"
    >
      <styled.div
        className="bp3-skeleton"
        key="data-table-item-empty-cell-name-9"
      >
         
      </styled.div>
    </styled.div>
    <styled.div
      key="data-table-empty-row-10"
    >
      <styled.div
        className="bp3-skeleton"
        key="data-table-item-empty-cell-name-10"
      >
         
      </styled.div>
    </styled.div>
    <styled.div
      key="data-table-empty-row-11"
    >
      <styled.div
        className="bp3-skeleton"
        key="data-table-item-empty-cell-name-11"
      >
         
      </styled.div>
    </styled.div>
    <styled.div
      key="data-table-empty-row-12"
    >
      <styled.div
        className="bp3-skeleton"
        key="data-table-item-empty-cell-name-12"
      >
         
      </styled.div>
    </styled.div>
    <styled.div
      key="data-table-empty-row-13"
    >
      <styled.div
        className="bp3-skeleton"
        key="data-table-item-empty-cell-name-13"
      >
         
      </styled.div>
    </styled.div>
    <styled.div
      key="data-table-empty-row-14"
    >
      <styled.div
        className="bp3-skeleton"
        key="data-table-item-empty-cell-name-14"
      >
         
      </styled.div>
    </styled.div>
    <styled.div
      key="data-table-empty-row-15"
    >
      <styled.div
        className="bp3-skeleton"
        key="data-table-item-empty-cell-name-15"
      >
         
      </styled.div>
    </styled.div>
    <styled.div
      key="data-table-empty-row-16"
    >
      <styled.div
        className="bp3-skeleton"
        key="data-table-item-empty-cell-name-16"
      >
         
      </styled.div>
    </styled.div>
    <styled.div
      key="data-table-empty-row-17"
    >
      <styled.div
        className="bp3-skeleton"
        key="data-table-item-empty-cell-name-17"
      >
         
      </styled.div>
    </styled.div>
    <styled.div
      key="data-table-empty-row-18"
    >
      <styled.div
        className="bp3-skeleton"
        key="data-table-item-empty-cell-name-18"
      >
         
      </styled.div>
    </styled.div>
    <styled.div
      key="data-table-empty-row-19"
    >
      <styled.div
        className="bp3-skeleton"
        key="data-table-item-empty-cell-name-19"
      >
         
      </styled.div>
    </styled.div>
    <styled.div
      key="data-table-empty-row-20"
    >
      <styled.div
        className="bp3-skeleton"
        key="data-table-item-empty-cell-name-20"
      >
         
      </styled.div>
    </styled.div>
    <styled.div
      key="data-table-empty-row-21"
    >
      <styled.div
        className="bp3-skeleton"
        key="data-table-item-empty-cell-name-21"
      >
         
      </styled.div>
    </styled.div>
    <styled.div
      key="data-table-empty-row-22"
    >
      <styled.div
        className="bp3-skeleton"
        key="data-table-item-empty-cell-name-22"
      >
         
      </styled.div>
    </styled.div>
    <styled.div
      key="data-table-empty-row-23"
    >
      <styled.div
        className="bp3-skeleton"
        key="data-table-item-empty-cell-name-23"
      >
         
      </styled.div>
    </styled.div>
    <styled.div
      key="data-table-empty-row-24"
    >
      <styled.div
        className="bp3-skeleton"
        key="data-table-item-empty-cell-name-24"
      >
         
      </styled.div>
    </styled.div>
    <styled.div
      key="data-table-empty-row-25"
    >
      <styled.div
        className="bp3-skeleton"
        key="data-table-item-empty-cell-name-25"
      >
         
      </styled.div>
    </styled.div>
    <styled.div
      key="data-table-empty-row-26"
    >
      <styled.div
        className="bp3-skeleton"
        key="data-table-item-empty-cell-name-26"
      >
         
      </styled.div>
    </styled.div>
    <styled.div
      key="data-table-empty-row-27"
    >
      <styled.div
        className="bp3-skeleton"
        key="data-table-item-empty-cell-name-27"
      >
         
      </styled.div>
    </styled.div>
    <styled.div
      key="data-table-empty-row-28"
    >
      <styled.div
        className="bp3-skeleton"
        key="data-table-item-empty-cell-name-28"
      >
         
      </styled.div>
    </styled.div>
    <styled.div
      key="data-table-empty-row-29"
    >
      <styled.div
        className="bp3-skeleton"
        key="data-table-item-empty-cell-name-29"
      >
         
      </styled.div>
    </styled.div>
    <styled.div
      key="data-table-empty-row-30"
    >
      <styled.div
        className="bp3-skeleton"
        key="data-table-item-empty-cell-name-30"
      >
         
      </styled.div>
    </styled.div>
    <styled.div
      key="data-table-empty-row-31"
    >
      <styled.div
        className="bp3-skeleton"
        key="data-table-item-empty-cell-name-31"
      >
         
      </styled.div>
    </styled.div>
    <styled.div
      key="data-table-empty-row-32"
    >
      <styled.div
        className="bp3-skeleton"
        key="data-table-item-empty-cell-name-32"
      >
         
      </styled.div>
    </styled.div>
    <styled.div
      key="data-table-empty-row-33"
    >
      <styled.div
        className="bp3-skeleton"
        key="data-table-item-empty-cell-name-33"
      >
         
      </styled.div>
    </styled.div>
    <styled.div
      key="data-table-empty-row-34"
    >
      <styled.div
        className="bp3-skeleton"
        key="data-table-item-empty-cell-name-34"
      >
         
      </styled.div>
    </styled.div>
    <styled.div
      key="data-table-empty-row-35"
    >
      <styled.div
        className="bp3-skeleton"
        key="data-table-item-empty-cell-name-35"
      >
         
      </styled.div>
    </styled.div>
    <styled.div
      key="data-table-empty-row-36"
    >
      <styled.div
        className="bp3-skeleton"
        key="data-table-item-empty-cell-name-36"
      >
         
      </styled.div>
    </styled.div>
    <styled.div
      key="data-table-empty-row-37"
    >
      <styled.div
        className="bp3-skeleton"
        key="data-table-item-empty-cell-name-37"
      >
         
      </styled.div>
    </styled.div>
    <styled.div
      key="data-table-empty-row-38"
    >
      <styled.div
        className="bp3-skeleton"
        key="data-table-item-empty-cell-name-38"
      >
         
      </styled.div>
    </styled.div>
    <styled.div
      key="data-table-empty-row-39"
    >
      <styled.div
        className="bp3-skeleton"
        key="data-table-item-empty-cell-name-39"
      >
         
      </styled.div>
    </styled.div>
    <styled.div
      key="data-table-empty-row-40"
    >
      <styled.div
        className="bp3-skeleton"
        key="data-table-item-empty-cell-name-40"
      >
         
      </styled.div>
    </styled.div>
    <styled.div
      key="data-table-empty-row-41"
    >
      <styled.div
        className="bp3-skeleton"
        key="data-table-item-empty-cell-name-41"
      >
         
      </styled.div>
    </styled.div>
    <styled.div
      key="data-table-empty-row-42"
    >
      <styled.div
        className="bp3-skeleton"
        key="data-table-item-empty-cell-name-42"
      >
         
      </styled.div>
    </styled.div>
    <styled.div
      key="data-table-empty-row-43"
    >
      <styled.div
        className="bp3-skeleton"
        key="data-table-item-empty-cell-name-43"
      >
         
      </styled.div>
    </styled.div>
    <styled.div
      key="data-table-empty-row-44"
    >
      <styled.div
        className="bp3-skeleton"
        key="data-table-item-empty-cell-name-44"
      >
         
      </styled.div>
    </styled.div>
    <styled.div
      key="data-table-empty-row-45"
    >
      <styled.div
        className="bp3-skeleton"
        key="data-table-item-empty-cell-name-45"
      >
         
      </styled.div>
    </styled.div>
    <styled.div
      key="data-table-empty-row-46"
    >
      <styled.div
        className="bp3-skeleton"
        key="data-table-item-empty-cell-name-46"
      >
         
      </styled.div>
    </styled.div>
    <styled.div
      key="data-table-empty-row-47"
    >
      <styled.div
        className="bp3-skeleton"
        key="data-table-item-empty-cell-name-47"
      >
         
      </styled.div>
    </styled.div>
    <styled.div
      key="data-table-empty-row-48"
    >
      <styled.div
        className="bp3-skeleton"
        key="data-table-item-empty-cell-name-48"
      >
         
      </styled.div>
    </styled.div>
    <styled.div
      key="data-table-empty-row-49"
    >
      <styled.div
        className="bp3-skeleton"
        key="data-table-item-empty-cell-name-49"
      >
         
      </styled.div>
    </styled.div>
  </styled.div>
  <styled.div>
    <styled.div
      flex={1}
      style={
        Object {
          "alignItems": "center",
          "justifyContent": "start",
        }
      }
    >
      <Blueprint3.Button
        icon="refresh"
        minimal={true}
        onClick={[Function]}
      />
      <styled.div>
        žiadny výber
      </styled.div>
    </styled.div>
    <styled.div
      flex={4}
    >
      <Pagination
        onChange={[Function]}
        page={1}
        size={50}
        total={0}
      />
    </styled.div>
    <styled.div
      flex={1}
    >
       
    </styled.div>
  </styled.div>
</styled.div>
`;
